cardano-sync/src/Cardano/Sync/LedgerState.hs

```diff
@@ -458,10 +440,14 @@ cleanupLedgerStateFiles env slotNo = do
 
 loadLedgerAtPoint :: LedgerEnv -> CardanoPoint -> Bool -> IO (Either [LedgerStateFile] CardanoLedgerState)
 loadLedgerAtPoint env point delFiles = do
+    -- Ledger states are growing to become very big in memory.
+    -- Before parsing the new ledger state we need to make sure the old ledger state
+    -- is or can be garbage collected.
+    writeLedgerState env Nothing
     mst <- findStateFromPoint env point delFiles
```

Comment:
maybe we want to force a gc at this point, since the old ledger state is a very big memory chunk.

Reply:
`performMajorGC`?